### ISSUE TITLE
sf::Http::Response resets field variables

### ISSUE THREAD

**Issue by diamondo25** [SFML/SFML#134]
Hey, I noticed that if you set multiple cookies with Set-Cookie, SFML will reset the field everytime a new one is read.

**Comment by binary1248** [comment]
Bump.

What's the status on this?

**Comment by Bromeon** [comment]
So the new API would look as follows?
```cpp
class Request
{
    void setField(const std::string& field, const std::string& value);
    void setFieldValues(const std::string& field, const std::vector<std::string>& values);
};

class Response
{
    const std::string&              getField(const std::string& field) const;
    const std::vector<std::string>& getFieldValues(const std::string& field) const;
};
```
The question is what `getField()` would return if there are multiple values.

Maybe we should discuss/implement the `sf::Message` base class and mixed get/set methods in a separate step, as it would also concern the other members, not just the field.

**Comment by vfjpl** [comment]
Bump.

What's the status on this?

**Comment by binary1248** [comment]
sf::Http isn't really worth using or maintaining considering that there are already better replacements for it and networking is scheduled to become standardized in the future.

Check [this list](https://github.com/fffaraz/awesome-cpp#networking) out for alternatives.

**Comment by vfjpl** [comment]
@binary1248 I modified Http.cpp for my needs. You can take a look at https://github.com/vfjpl/oldmargonem/blob/master/Http.cpp and check if there is something interesting for upstream.
https://github.com/vfjpl/oldmargonem/blob/ad2d16a38160a0253f50c55db967140226eb32f6/Http.cpp

**Comment by Bromeon** [comment]
> 
> 
> sf::Http isn't really worth using or maintaining considering that there are already better replacements for it and networking is scheduled to become standardized in the future.
> 
> Check [this list](https://github.com/fffaraz/awesome-cpp#networking) out for alternatives.

I agree, many other libraries (or even languages) provide much more functionality regarding networking. Nowadays, HTTPS is pretty much standard everywhere (with browsers actively discouraging HTTP), and many clients provide readily available features for Basic Auth, Cookies, JSON encoding/decoding, etc.

Should this be closed as wontfix, or do we still want to implement a simple solution like the one I proposed [here](https://github.com/SFML/SFML/issues/134#issuecomment-160432767)?

**Comment by kennyrkun** [comment]
The HTTP capabilities of SFML were a huge part in deciding to use the library for me, so I would really appreciate continued support on this part of the library.

Doesn't need to be able to compete with bigger libraries, but I would love a simple, solid, base framework to work with HTTP/S, just like the rest of the library :)

**Comment by JonnyPtn** [comment]
I'd think we want to keep this alive given we are continuing to support and update the network module (rightly so in my opinion)

While I'm no network expert, reading the latest spec here https://www.rfc-editor.org/rfc/rfc9110.html#name-field-order It seems the situation has changed from the previous discussions here. My understanding from reading that is that we should **not** allow multiple fields with the same name, except `Set-Cookie` which is allowed as an exception.

With that in mind, perhaps keep the existing API for header fields (perhaps warn/error if a field already exists) and add a separate method for setting/getting cookies?
